cardano-db-sync/src/Cardano/DbSync.hs

```diff
@@ -238,50 +236,55 @@ runDbSyncNodeNodeClient iomgr trce plugin nodeConfig (SocketPath socketPath) = d
 
 
 localInitiatorNetworkApplication
-  :: forall blk.
-     (blk ~ ByronBlock)
-  -- TODO: the need of a 'Proxy' is an evidence that blk type is not really
-  -- needed here.  The wallet client should use some concrete type of block
-  -- from 'cardano-chain'.  This should remove the dependency of this module
-  -- from 'ouroboros-consensus'.
-  => Trace IO Text
+  :: NodeToClientVersion
+  -> Trace IO Text
   -> DbSyncNodePlugin
   -> TopLevelConfig ByronBlock
-  -> StrictTMVar IO (GenTx blk)
+  -> StrictTMVar IO (GenTx ByronBlock)
   -> ConnectionId LocalAddress
   -> OuroborosApplication 'InitiatorApp BSL.ByteString IO () Void
-localInitiatorNetworkApplication trce plugin pInfoConfig txv _ =
-    nodeToClientProtocols NodeToClientProtocols
-      { localChainSyncProtocol =
-          -- local chain sync protocol
-          (InitiatorProtocolOnly $ MuxPeerRaw $ \channel ->
-            liftIO . logException trce "ChainSyncWithBlocksPtcl: " $ do
-              logInfo trce "Starting chainSyncClient"
-              latestPoints <- getLatestPoints
-              currentTip <- getCurrentTipBlockNo
-              logDbState trce
-              actionQueue <- newDbActionQueue
-              (metrics, server) <- registerMetricsServer
-              race_
-                (runDbThread trce plugin metrics actionQueue)
-                (runPipelinedPeer
-                      localChainSyncTracer (localChainSyncCodec @blk pInfoConfig) channel
-                      (chainSyncClientPeerPipelined (chainSyncClient trce metrics latestPoints currentTip actionQueue))
-                    )
-              atomically $
-                writeDbActionQueue actionQueue DbFinish
-              cancel server)
-        , localTxSubmissionProtocol =
-            -- local tx submission
-            (InitiatorProtocolOnly $ MuxPeer
-              (contramap (Text.pack . show) . toLogObject $ appendName "db-sync-local-tx" trce)
-              localTxSubmissionCodec
-              (localTxSubmissionClientPeer (txSubmissionClient txv)))
-        }
+localInitiatorNetworkApplication nodeToClientVersion trce plugin pInfoConfig txv _ =
+    nodeToClientProtocols
+       (NodeToClientProtocols {
+           localChainSyncProtocol = localChainSyncProtocol
+          ,localTxSubmissionProtocol = localTxSubmissionProtocol
+          ,localStateQueryProtocol = Prelude.error "localTxSubmissionProtocol" -- todo!
```

Comment:
TODO: use the ready-made null protocol

Reply:
We have a null peer now: https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-network/src/Ouroboros/Network/NodeToClient.hs#L730

Reply:
Yes, it is used now